include/SDL3/SDL_test_common.h

```diff
@@ -55,10 +55,10 @@ typedef Uint32 SDLTest_VerboseFlags;
 
 /* !< Function pointer parsing one argument at argv[index], returning the number of parsed arguments,
  *    or a negative value when the argument is invalid */
-typedef int (*SDLTest_ParseArgumentsFp)(void *data, char **argv, int index);
+typedef int (SDLCALL *SDLTest_ParseArgumentsFp)(void *data, char **argv, int index);
```

Comment:
You should mark any `SDLTest_ParseArgumentsFp` type functions as `SDLCALL` -- same for others

Reply:
Fixed, I think.

I'm thinking about building [cmake/test](https://github.com/libsdl-org/SDL/tree/main/cmake/test) with fastcall on x86 MSVC.
Would such a thing be useful?

Reply:
This seems to be missing, which isn't actually related to marking SDL_test procs as SDLCALL:

```diff
diff --git a/src/test/SDL_test_common.c b/src/test/SDL_test_common.c
index e749e5c..9b67687 100644
--- a/src/test/SDL_test_common.c
+++ b/src/test/SDL_test_common.c
@@ -1971,7 +1971,7 @@ typedef struct
     size_t size;
 } SDLTest_ClipboardData;
 
-static void SDLTest_ScreenShotClipboardCleanup(void *context)
+static void SDLCALL SDLTest_ScreenShotClipboardCleanup(void *context)
 {
     SDLTest_ClipboardData *data = (SDLTest_ClipboardData *)context;
 
@@ -1983,7 +1983,7 @@ static void SDLTest_ScreenShotClipboardCleanup(void *context)
     SDL_free(data);
 }
 
-static const void *SDLTest_ScreenShotClipboardProvider(void *context, const char *mime_type, size_t *size)
+static const void * SDLCALL SDLTest_ScreenShotClipboardProvider(void *context, const char *mime_type, size_t *size)
 {
     SDLTest_ClipboardData *data = (SDLTest_ClipboardData *)context;
 
```

Reply:
Also, any satellite libs using SDL_test procs will need updating accordingly.

Reply:
> This seems to be missing, which isn't actually related to marking SDL_test procs as SDLCALL:

Thanks! Fixed.


> Building SDL_test as fastcall for x86 should be enough to detect any missing procs not marked as SDLCALL: should trigger warnings about calling convention mismatches.
> 
> Building SDL itself as fastcalll won't help with anything I think.

I'm looking at this from a ci perspective.
Building everything in fastcall will give the code an extra workout :)

Reply:
> Building SDL_test as fastcall for x86 should be enough to detect any missing procs not marked as SDLCALL: should trigger warnings about calling convention mismatches.
> 
> Building SDL itself as fastcalll won't help with anything I think.

Looks like SDL3 does not build with a non-default (non-cdecl) calling convention
https://github.com/madebr/SDL/actions/runs/10736750323/job/29776894241#step:23:90

Reply:
> Looks like SDL3 does not build with a non-default (non-cdecl) calling convention
> https://github.com/madebr/SDL/actions/runs/10736750323/job/29776894241#step:23:90

Eh? Looks like prototype doesn't dictate calling convention on the procedure's definition for msvc?

In any case, missing SDLCALL has always been a problem with SDL, so I'd say don't worry about it much for CI. Only watcom builds used to exposed those (the one I noted above is found that way.)

Reply:
The issue starts with `SDL_main`, so users would have to add `SDLCALL` to their `main`.
https://github.com/madebr/SDL/actions/runs/10738079225/job/29781014809#step:28:178